arangod/RestServer/TtlFeature.cpp

```diff
@@ -332,7 +332,7 @@ class TtlThread final : public ServerThread<ArangodServer> {
         }
 
         if (ServerState::instance()->isDBServer() &&
-            !collection->followers()->getLeader().empty()) {
+            !collection->isLeadingShard()) {
```

Comment:
The `isLeadingShard` return `false` if the collection `isAStub()`. I'm pointing it out, although as far as I know that can only happen on a coordinator.

Reply:
Good Spot!
That is actually expected and I think the correct behaviour.
isAStub indicates we are talking to the "Cluster-Wide" Collection, that does not have data itself, but only delegates to shards.
This collections cannot do any of the below operations, so we are good, and have another layer of protection.